opentracing-spring-extension/opentracing-spring-web-extension-autoconfigure/src/test/java/org/zalando/opentracing/spring/web/extension/autoconfigure/OpenTracingWebExtensionAutoConfigurationTest.java

```diff
@@ -0,0 +1,134 @@
+package org.zalando.opentracing.spring.web.extension.autoconfigure;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.interceptor.HandlerInterceptorSpanDecorator;
+import io.opentracing.contrib.spring.web.interceptor.TracingHandlerInterceptor;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.client.DefaultResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.util.Collections.emptyList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasEntry;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.DEFINED_PORT;
+import static org.springframework.http.HttpStatus.OK;
+
+@ExtendWith(SpringExtension.class) // only needed for Spring 4 support
+@SpringBootTest(
+        webEnvironment = DEFINED_PORT,
```

Comment:
`RANDOM_PORT` and inject port number? I had troubles with tests on `8080` in the past

Reply:
> I had troubles with tests on `8080` in the past

Like what?

> `RANDOM_PORT` and inject port number?

I'd delay that until I see an issue. Injecting the port number adds even more boilerplate to the test.